nomad/drainer/drainer.go

```diff
@@ -295,13 +295,11 @@ func (n *NodeDrainer) handleMigratedAllocs(allocs []*structs.Allocation) {
 	}
 	n.l.RUnlock()
 
-	// TODO(alex) This should probably be a single Raft transaction
-	for _, doneNode := range done {
-		index, err := n.raft.NodeDrainComplete(doneNode)
-		if err != nil {
-			n.logger.Printf("[ERR] nomad.drain: failed to unset drain for node %q: %v", doneNode, err)
-		} else {
-			n.logger.Printf("[INFO] nomad.drain: node %q completed draining at index %d", doneNode, index)
+	// Submit the node transistions in a sharded form to ensure a reasonable
+	// Raft transaction size.
+	for _, nodes := range partitionIds(done) {
```

Comment:
I'd rather this go into the shim since it's a raft-layer-implementation-detail and we're passing a slice to the shim anyway.